sql/log.h

```diff
@@ -1066,10 +1066,14 @@ int query_error_code(THD *thd, bool not_killed);
 uint purge_log_get_error_code(int res);
 
 int vprint_msg_to_log(enum loglevel level, const char *format, va_list args);
-void sql_print_error(const char *format, ...);
-void sql_print_warning(const char *format, ...);
-void sql_print_information(const char *format, ...);
-typedef void (*sql_print_message_func)(const char *format, ...);
+void sql_print_error(const char *format, ...)
+                     ATTRIBUTE_FORMAT_FPTR(printf, 1, 2);
+void sql_print_warning(const char *format, ...)
+                       ATTRIBUTE_FORMAT_FPTR(printf, 1, 2);
+void sql_print_information(const char *format, ...)
+                           ATTRIBUTE_FORMAT_FPTR(printf, 1, 2);
```

Comment:
@vuvova pointed out that these functions use my_vsnprintf(), which has extended format. See include/mysql/service_my_snprintf.h for description.

Said the above I believe we can't use ATTRIBUTE_FORMAT() for these functions. :(